sharmavikas4/MERN_BLOG
# manage error-handling using try_catch blocks ans make functions async…

… wehere needed for proper functioning


Here’s a sample of your Pull Request description filled out in the format you’ve provided:

What does this PR do?
This PR introduces error handling across various routes and refactors asynchronous functions to use async/await, enhancing stability and code readability. It addresses issues where unhandled exceptions could crash the application and improves the maintainability of the code.

Fixes # (issue number, if applicable)

Key Changes:
Added error handling with try/catch blocks to manage exceptions in API routes.
Refactored routes to use async/await, replacing callback-based asynchronous operations for improved control flow.
Improved error handling for user authentication middleware.
Enhanced Express server with error handling middleware for better request/response management.

Type of change
1. Bug fix
2. Chore (refactoring code, technical debt, workflow improvements)


How should this be tested?
 Test A: Test the API endpoints like /dashboard and /likedPosts to ensure that they handle errors properly and still return expected results.
 Test B: Check user authentication flow to verify that unauthenticated users receive correct error messages.
